### ISSUE TITLE
Not all the dataloaders in multi-task learning are trained

### ISSUE THREAD

**Issue by CarinaXZZ** [UKPLab/sentence-transformers#1125]
Hi,
I'm trying to apply multi-task learning with the example you provided. And I created 3 different data loaders and loss functions. However, while training the model, I notice that only the smallest Dataloader is trained. For example, the log shows that only STS-B is trained in the example you provided.
![image](https://user-images.githubusercontent.com/29379208/130066932-53c6fa94-cead-44ac-92f6-9b8b2be27fa6.png)

In my case, I applied:
```
model.fit(train_objectives=[(train_dataloader_MultipleNegativesRankingLoss, train_loss_MultipleNegativesRankingLoss), (train_dataloader_ConstrativeLoss, train_loss_ConstrativeLoss)],
          evaluator=seq_evaluator,
          epochs=num_epochs,
          warmup_steps=1000,
          output_path=model_save_path
          )
```
But only the constrativeLoss is trained. Can you help me with this? Thanks.

**Comment by nreimers** [comment]
How do you see that only the constrativeLoss is trained?

Data is loaded from all dataloaders. However, 1 epoch = steps for the smallest data loader. But in each step, data from all 3 data loaders are loaded.

**Comment by CarinaXZZ** [comment]
> How do you see that only the constrativeLoss is trained?
> 
> Data is loaded from all dataloaders. However, 1 epoch = steps for the smallest data loader. But in each step, data from all 3 data loaders are loaded.

I see, thank you! So the number of batches depends on the smallest dataset, in my case, ConstrativeLoss is the smallest dataset.
If my two datasets are very different in size (391k samples for Multiple Negatives Ranking and 10k for Online Constractive), is there any other way to train it apart from increasing epochs?

**Comment by nreimers** [comment]
You can pass the parameter how many train steps each epoch has.
